airbyte-cdk/java/airbyte-cdk/db-sources/src/main/kotlin/io/airbyte/cdk/integrations/source/jdbc/AbstractJdbcSource.kt

```diff
@@ -270,9 +271,44 @@ abstract class AbstractJdbcSource<Datatype>(
                     CheckedFunction { connection: Connection -> connection.metaData.catalogs },
                     CheckedFunction { queryResult: ResultSet ->
                         sourceOperations.rowToJson(queryResult)
-                    }
+                    },
                 )
-            }
+            },
+            CheckedConsumer { database: JdbcDatabase ->
+                var schemas = ArrayList<String>()
+                if (config!!.has(JdbcUtils.SCHEMAS_KEY) && config[JdbcUtils.SCHEMAS_KEY].isArray) {
+                    for (schema in config[JdbcUtils.SCHEMAS_KEY]) {
+                        schemas.add(schema.asText())
+                    }
+                }
+                // if UI has schemas specified, check if the user has select access to any table
+                if (schemas.isNotEmpty()) {
+                    for (schema in schemas) {
+                        LOGGER.info {
+                            "Checking if the user can perform select to any table in schema: $schema"
+                        }
+                        val tablesOfSchema = database.metaData.getTables(null, schema, "%", null)
+                        if (tablesOfSchema.next()) {
+                            var privileges =
+                                getPrivilegesTableForCurrentUser<JdbcPrivilegeDto>(database, schema)
+                            if (privileges.isEmpty()) {
+                                LOGGER.info {
+                                    "No table from schema $schema is accessible for the user."
+                                }
+                                throw ConfigErrorException(
+                                    "User can not perform select to any table from schema: $schema"
+                                )
+                            }
+                        } else {
+                            LOGGER.info { "Schema $schema does not contain any table." }
+                        }
+                    }
+                } else {
+                    LOGGER.info {
+                        "No schema has been provided at the moment, skip table permission check."
```

Comment:
@postamar, on this condition, I still prefer a no-op as, in this case, the user really hasn't provided any schema. Actually, I found that the user has to intentionally delete the default schema our UI filled to run into this condition. In such a case, I think the user may not be surprised if no table shows up.

Reply:
That makes sense.